File: 872-leaf-similar-trees/872-leaf-similar-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool leafSimilar(TreeNode* root1, TreeNode* root2) 
    {
        vector<int> leaves1;
        vector<int> leaves2;
        dfs(root1,leaves1);
        dfs(root2,leaves2);
        
        return leaves1 == leaves2;
    }
    void dfs(TreeNode* root1,vector<int> &leaves1)
    {
        if(!root1)
            return;
        if(root1 -> left == NULL && root1 -> right == NULL)
            leaves1.push_back(root1 -> val);
        dfs(root1 -> left,leaves1);
        dfs(root1 -> right,leaves1);   
    }
};

File: 872-leaf-similar-trees/872-leaf-similar-trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var leafSimilar = function(root1, root2) {
    let dfs = function(root,leaves){
        if(!root)
            return;
        if(root.left == null && root.right == null)
            leaves.push(root.val);
        dfs(root.left,leaves);
        dfs(root.right,leaves);
    };
    
    let leaves1 = [];
    let leaves2 = [];
    dfs(root1,leaves1);
    dfs(root2,leaves2);
     return (leaves1.length == leaves2.length &&
            leaves1.every((v, i) => v === leaves2[i]));
};